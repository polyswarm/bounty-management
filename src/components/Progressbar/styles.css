.Progressbar {
  border-radius: 12px;
  background-color: #dedede;
  height: 24px;
}

.Progressbar-Fill {
  border-radius: 12px;
  background-color: #8500ff;
  height: 24px;
}
