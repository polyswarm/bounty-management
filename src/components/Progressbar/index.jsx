// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';

// Component Imports
import './styles.css';

class Progressbar extends Component {

  render() {
    const { props: { progress } } = this;
    let p = progress;
    if (!p || p < 0) {
      p = 0;
    } else if (p > 100) {
      p = 100;
    }
    const style = {width: `${p}%`};
    return (
      <div { ...this.props} className='Progressbar'>
        <div className='Progressbar-Fill' style={style} />
      </div>
    );
  }
}

Progressbar.propTypes = {
  progress: PropTypes.number
};

export default Progressbar;
