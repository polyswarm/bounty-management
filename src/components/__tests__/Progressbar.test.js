import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson} from 'enzyme-to-json';
import Progressbar from '../Progressbar';

it('renders without crashing', () => {
  const wrapper = render(<Progressbar />);
  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('sets progress width according to value passed', () => {
  const wrapper = mount(<Progressbar progress={40}/>);

  expect(wrapper.find('.Progressbar-Fill').props().style.width).toBe('40%');
});

it('caps progress at 100%', () => {
  const wrapper = mount(<Progressbar progress={10000}/>);

  expect(wrapper.find('.Progressbar-Fill').props().style.width).toBe('100%');
});

it('uses 0 as the minimum', () => {
  const wrapper = mount(<Progressbar progress={-10}/>);

  expect(wrapper.find('.Progressbar-Fill').props().style.width).toBe('0%');
});

it('uses 0 when no progress given', () => {
  const wrapper = mount(<Progressbar/>);

  expect(wrapper.find('.Progressbar-Fill').props().style.width).toBe('0%');
});
