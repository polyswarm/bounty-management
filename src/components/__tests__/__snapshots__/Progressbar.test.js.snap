// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="Progressbar"
>
  <div
    class="Progressbar-Fill"
    style="width:0%"
  >
    
  </div>
</div>
`;
